doc/source/reference/global_state.rst

```diff
@@ -71,3 +70,10 @@ and set the ``ndarray.base``.
 
 .. versionchanged:: 1.25.2
     This variable is only checked on the first import.
+
+Legacy User DTypes
```

Comment:
This does not feel very logical here. Better at the end of thread safety? Also, isn't this only an issue if one defines legacy user types, which I'd think very few programs do?

Reply:
I could combine the two pages maybe? 

It's here because it's global state. 

I agree it's extremely niche and it's a legacy feature that we aren't really planning to advocate for people to use going forward. I could also not mention it, no one has cared or noticed up until now.

Reply:
We could put the info in `PyArray_RegisterDataType` and link from the free-threaded section?
(Right now the user-dtype won't be compiled with free-threaded support anyway, but it is tempting of course...  And you probably get away with it since most will be probably be added at import time when no threads are active.)

I don't think it needs to be here really.  It is global state, but it modifies NumPy runtime behavior very explicitly (and ideally not at all unless you use that dtype).
(I.e. maybe the global state name is not great but not sure what is better.  "Global config"?)